Sources/Deferred.swift

```diff
@@ -91,18 +91,26 @@ public struct Deferred<Value>: FutureType, PromiseType {
     :returns: The determined value, if filled within the timeout, or `nil`.
     */
     public func wait(time: Timeout) -> Value? {
-        let queue = dispatch_get_global_queue(QOS_CLASS_UTILITY, 0)
-        var value: Value?
-        let handler = upon(queue, options: DISPATCH_BLOCK_ENFORCE_QOS_CLASS) {
-            value = $0
+        var result: Value?
+        func assign(value: Value) {
+            result = value
+        }
+
+        // Pure FutureType implementations implement isFilled in terms of wait()
```

Comment:
This comment implies the next line would be infinitely circular, wouldn't it? If wait calls isFilled and isFilled is implemented in terms of wait.